packages/workers-shared/asset-worker/src/assets-manifest.ts

```diff
@@ -96,3 +96,11 @@ const contentHashToKey = (buffer: Uint8Array) => {
 	).slice(0, CONTENT_HASH_SIZE);
 	return [...contentHash].map((b) => b.toString(16).padStart(2, "0")).join("");
 };
+
+const encodeFilePath = (filePath: string) => {
+	const encodedPath = filePath
+		.split("/")
+		.map((segment) => encodeURIComponent(segment))
+		.join("/");
```

Comment:
Why do we need to split on `/`? Does calling `encodeURIComponent` on the whole path not work?

Reply:
```
> encodeURI('/about/fünky:path.txt')
'/about/f%C3%BCnky:path.txt'
> encodeURIComponent('/about/fünky:path.txt')
'%2Fabout%2Ff%C3%BCnky%3Apath.txt'
> '/about/fünky:path.txt'.split('/').map(x => encodeURIComponent(x)).join('/')
'/about/f%C3%BCnky%3Apath.txt'
> '/about/fünky:path.txt'.split('/').map(x => encodeURIComponent(x)).join('/') === encodeURI('/about/fünky:path.txt')
false
```

Reply:
Considering this needs to match how the asset manifest is constructed, it would be great if this could be colocated with that

Reply:
Sorry, I don't think I was super clear—I know it's different, but I'm wondering why we can't just encode the whole thing?

Reply:
As in, just `encodeURIComponent(path)` and only storing the encoded paths. Why do we need slashes specifically to be preserved?

Reply:
Seems like this manifest is sent to an API endpoint so I imagine the endpoint itself has made this decision https://github.com/cloudflare/workers-sdk/blob/main/packages/wrangler/src/experimental-assets.ts#L65

If we want to change this, we would need to do so before users are uploading assets because the blob has these keys persisted remotely

Reply:
Hopefully we still can, since we're pre release. Cc @GregBrimble

Reply:
Currently EWC is checking for a `/`at the start of manifest keys but Greg says he’s happy to change that? So I’ll tidy this PR up in a bit and encode the whole thing as suggested :)

Reply:
One more. Is there a reason we can't use `encodeURI` instead of `encodeURIComponent`? Might make reading the manifest by a human slightly easier if the slashes aren't also encoded eg `/file-1.txt` instead of `%2Ffile-1.txt`

Reply:
Another question: why do we need to encode these paths before we hash them?

Reply:
EncodeURI doesn't encode characters like '?' and '#' while encodeURIComponent does. We decided to allow users to upload filenames with those characters if they really wanted to do that - (e.g. the file `foo?bar.txt` would be accessible at `/foo%3Fbar.txt`). To allow that, we need use encodeURIComponent and also encode before hashing so that the hashed path corresponds to `/foo%3Fbar.txt`.